### PR TITLE
[BEAM-12645] Fix code-cov flakes due to monorepo.

### PR DESCRIPTION
**Pull request by lostluck**
Use features of codecov to avoid overwriting codecov.
- Flag the different uploads (go and python) so they are merged, and not clobbered.
- Update the Go SDK upload to use the github action, rather than the decommissioned bash script.

In particular, Go SDK coverage is presently lower than python, and Go uploads would overwrite a previous python upload entirely. Codecov has a [flags](https://docs.codecov.com/docs/flags) mechnaism for this that segments uploads into their own groups, and merges everything at the top level.

Further, we likely want to have carryforward enabled, since not all tests are run on every run, avoiding weird "0%" coverage messages, when very little was changed.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @aaltay

**Comment by lostluck** [comment]
For this, I'm on the side of "lets get it in" since that's probably easier to tell if it's having the desired effect. I don't think the codecov.yml changes will take effect until we do anyway.

File: .github/codecov.yml [hunk 1]
```diff
@@ -59,3 +63,7 @@ ignore:
   - "**/*_test.py"
   - "**/*_test_py3*.py"
   - "**/*_microbenchmark.py"
+
+flag_management:
+  default_rules: # the rules that will be followed for any flag added, generally
```
Comment (aaltay) [inline .github/codecov.yml:68]: Would it help to add a link to the codecov docs for these flags?

**Review by aaltay — APPROVED** [review]
Thank you!

Is there a way to verify it? Or do we have to merge and check data?

**Pull request merged**